src/ripple/app/ledger/LedgerToJson.h

```diff
@@ -129,49 +132,84 @@ void fillJson (Object& json, LedgerFill const& fill)
             {
                 if (type == SHAMapTreeNode::tnTRANSACTION_NM)
                 {
-                    SerializerIterator sit (item->peekSerializer ());
-                    STTx txn (sit);
-                    txns.append (txn.getJson (0));
+                     if (bBinary)
+                    {
+                        auto&& obj = appendObject (txns);
+                        obj[jss::tx_blob] = strHex (item->peekData ());
+                    }
```

Comment:
It seems as if the future name for this concept is "forwarding reference".

My intuition of what a forwarding reference "means" needs to be honed by experience.   One informal way to think of it is as "a reference that sometimes needs a destructor".

For a moment I was excited about the `<<` idea - but how would it work to set keyed values and such?

I am experimenting with a unified system where `Object` and `Array` are hidden under one `Union` type whose interface allows both `append` and `[]` indexing (though it throws if you try to append to an object or set a keyed value in an array.

It regularizes the interface a little more, at the cost of a tiny bit of protection from the type system, but I'm realizing it's beyond the scope of this - and it doesn't fix the forwarding reference issue.